Orm/Xtensive.Orm/Orm/Linq/Translator.Expressions.cs

```diff
@@ -207,6 +207,30 @@ protected override Expression VisitBinary(BinaryExpression binaryExpression)
           right = Visit(binaryExpression.Right);
         }
       }
+      // Following two checks for enums are here to improve result query
+      // performance because they let not to cast columns to integer.
+      else if (EnumRewritableOperations(binaryExpression) 
```

Comment:
Why not just have a single `else if` statement for the `EnumRewritableOperations` case? In that case, you could declare four variables as I suggested for the `PartialIndexFilterBuilder` inside that `else if` block. The cost of that is a duplicated block
```csharp
      else {
        left = Visit(binaryExpression.Left);
        right = Visit(binaryExpression.Right);
      }
```
but as for me, it is worth it.
Also please pay attention to the `EnumRewritableOperations` method implementation. We can avoid allocations there.